Tests/Unit/Hooks/HookServiceTest.php

```diff
@@ -0,0 +1,159 @@
+<?php
+declare(strict_types = 1);
+
+namespace OliverKlee\Seminars\Tests\Unit\Hooks;
+
+use Nimut\TestingFramework\TestCase\UnitTestCase;
+use OliverKlee\Seminars\Hooks\HookService;
+use OliverKlee\Seminars\Interfaces\Hook;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestHookInterface;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestHookImplementor;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Test case.
+ *
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+class HookServiceTest extends UnitTestCase
+{
+    /**
+     * @var array backed-up extension configuration of the TYPO3 configuration
+     *            variables
+     */
+    protected $extConfBackup = [];
+
+    protected function setUp()
+    {
+        $this->extConfBackup = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'];
+    }
+
+    protected function tearDown()
+    {
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'] = $this->extConfBackup;
+    }
+
+    /*
+     * Utility functions
+     */
+
+    /**
+     * Creates a TestHook implementor object.
+     *
+     * @return \OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestHookImplementor
+     */
+    protected function createTestHookImplementor()
+    {
+        return GeneralUtility::makeInstance(TestHookImplementor::class);
+    }
+
+    /**
+     * Creates a TestHook accepting Hook object.
+     *
+     * @return \OliverKlee\Seminars\Hooks\HookService
+     */
+    protected function createHookObject()
```

Comment:
Pleas make the non-public methods in the tests private. This makes it easier for static code analysis to find unused methods that can be removed.